src/io/parquet/reader/spatial_filter.rs

```diff
@@ -0,0 +1,279 @@
+    pub fn get_bbox(&self, rg_meta: &RowGroupMetaData) -> Result<Rect> {
+        let (minx, _) = parse_statistics_f64(rg_meta.column(self.minx_col))?;
+        let (miny, _) = parse_statistics_f64(rg_meta.column(self.miny_col))?;
+        let (_, maxx) = parse_statistics_f64(rg_meta.column(self.maxx_col))?;
+        let (_, maxy) = parse_statistics_f64(rg_meta.column(self.maxy_col))?;
+        Ok(Rect::new(
+            coord! { x: minx, y: miny },
+            coord! {x: maxx, y: maxy},
+        ))
+    }
+
+    /// Extract the bounding boxes for a sequence of row groups
+    pub fn get_bboxes(&self, row_groups: &[RowGroupMetaData]) -> Result<RectArray> {
+        let mut builder = RectBuilder::with_capacity(row_groups.len(), Default::default());
+        for rg_meta in row_groups.iter() {
+            builder.push_rect(Some(&self.get_bbox(rg_meta)?));
+        }
+        Ok(builder.finish())
+    }
+}
+
+pub(crate) fn apply_bbox_row_groups<T>(
+    builder: ArrowReaderBuilder<T>,
+    bbox_cols: ParquetBboxStatistics,
+    bbox_query: Rect,
+) -> Result<ArrowReaderBuilder<T>> {
+    let row_groups = builder.metadata().row_groups();
+    let row_groups_bounds = bbox_cols.get_bboxes(row_groups)?;
+    let mut intersects_row_groups_idxs = vec![];
+    for (row_group_idx, row_group_bounds) in row_groups_bounds.iter_values().enumerate() {
+        if rect_intersects(&row_group_bounds, &bbox_query) {
+            intersects_row_groups_idxs.push(row_group_idx);
+        }
+    }
+
+    Ok(builder.with_row_groups(intersects_row_groups_idxs))
+}
+
+pub(crate) fn apply_bbox_row_filter<T>(
+    builder: ArrowReaderBuilder<T>,
+    bbox_cols: ParquetBboxStatistics,
+    bbox_query: Rect,
+) -> Result<ArrowReaderBuilder<T>> {
+    let parquet_schema = builder.parquet_schema();
+    let predicate = construct_predicate(parquet_schema, bbox_cols, bbox_query)?;
+    let filter = RowFilter::new(vec![predicate]);
+    Ok(builder.with_row_filter(filter))
+}
+
+pub(crate) fn construct_predicate(
+    parquet_schema: &SchemaDescriptor,
+    bbox_cols: ParquetBboxStatistics,
+    bbox_query: Rect,
+) -> Result<Box<dyn ArrowPredicate>> {
+    let mask = ProjectionMask::leaves(
+        parquet_schema,
+        [
+            bbox_cols.minx_col,
+            bbox_cols.miny_col,
+            bbox_cols.maxx_col,
+            bbox_cols.maxy_col,
+        ],
+    );
+
+    let predicate = ArrowPredicateFn::new(mask, move |batch| {
+        let struct_col = batch.column(0).as_struct();
+
+        ///////////////////////////////////////////////
+        // TODO: come back to this
+        ///////////////////////////////////////////////
+
+        let minx_col = struct_col.column(0).as_primitive::<Float64Type>();
+        let maxx_col = struct_col.column(1).as_primitive::<Float64Type>();
+        let miny_col = struct_col.column(2).as_primitive::<Float64Type>();
+        let maxy_col = struct_col.column(3).as_primitive::<Float64Type>();
```

Comment:
Here this _happened_ to be in the Overture data that the order of columns within the struct was `minx, maxx, miny, maxy`. But the key is that the order of this struct column is the order of the _data_ not the order of the "mask query"

Reply:
I'll merge and come back to this. Tracked in https://github.com/geoarrow/geoarrow-rs/issues/593